src/gui/layertree/qgslayertreeview.cpp

```diff
@@ -154,7 +159,16 @@ void QgsLayerTreeView::modelRowsInserted( const QModelIndex &index, int start, i
         if ( QgsLayerTreeEmbeddedWidgetProvider *provider = QgsGui::layerTreeEmbeddedWidgetRegistry()->provider( providerId ) )
         {
           QModelIndex index = layerTreeModel()->legendNode2index( legendNodes[i] );
-          setIndexWidget( index, provider->createWidget( layer, i ) );
+          QWidget *wdgt = provider->createWidget( layer, i );
+          // Since column is resized to contents, limit the expanded width of embedded widgets,
+          // if they are not already limited, e.g. have the defaut MAX value.
```

Comment:
```suggestion
          // if they are not already limited, e.g. have the default MAX value.
```

Reply:
Oops, thanks. The spell check part of prepare commit script is failing for me on macOS.